src/test/java/games/strategy/engine/framework/startup/login/V2AuthenticatorTest.java

```diff
@@ -0,0 +1,242 @@
+package games.strategy.engine.framework.startup.login;
+
+import static com.googlecode.catchexception.CatchException.catchException;
+import static com.googlecode.catchexception.CatchException.caughtException;
+import static com.googlecode.catchexception.apis.CatchExceptionHamcrestMatchers.hasMessageThat;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.emptyString;
+import static org.hamcrest.Matchers.hasKey;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.nullValue;
+import static org.hamcrest.collection.IsMapContaining.hasEntry;
+import static org.junit.Assert.assertThat;
+
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.Map;
+
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+
+import games.strategy.engine.framework.startup.login.V2Authenticator.ChallengePropertyNames;
+import games.strategy.engine.framework.startup.login.V2Authenticator.ResponsePropertyNames;
+
+public final class V2AuthenticatorTest {
+  private static final String PASSWORD = "←PASSWORD↑WITH→UNICODE↓CHARS";
+
+  @Test
+  public void canProcessChallenge_ShouldReturnTrueWhenAllPropertiesPresent() {
+    final Map<String, String> challenge = newChallengeWithAllProperties();
+
+    assertThat(V2Authenticator.canProcessChallenge(challenge), is(true));
+  }
+
+  private static Map<String, String> newChallengeWithAllProperties() {
+    return Maps.newHashMap(ImmutableMap.of(
+        ChallengePropertyNames.NONCE, newBase64String(),
+        ChallengePropertyNames.SALT, newBase64String()));
+  }
+
+  private static String newBase64String() {
+    final byte[] bytes = new byte[8];
+    Arrays.fill(bytes, (byte) 1);
```

Comment:
Weird, I could have sworn this was possible. 
Maybe I just thought because it worked with f, d and l it would work with s and b too :-/
They should really add that